tools/cmake/mbed_greentea.cmake

```diff
@@ -55,9 +55,24 @@ macro(mbed_greentea_add_test)
             ${MBED_GREENTEA_TEST_SOURCES}
     )
 
-    if(MBED_BAREMETAL_GREENTEA_TEST)
-        list(APPEND MBED_GREENTEA_TEST_REQUIRED_LIBS mbed-baremetal mbed-greentea)
-    else()
+    # The CMake MBED_TEST_LINK_LIBRARIES command-line argument for selecting "mbed-baremetal" library
+    # or to add extra dependent libraries for any greentea test.
+    # For example:
+    #  - To select baremetal library, use cmake with -DMBED_TEST_LINK_LIBRARIES=mbed-baremetal
+    #  - To add extra error logging library to the test, use cmake with -DMBED_TEST_LINK_LIBRARIES=ext-errorlogging
+    #  - To select baremetal with extra external error logging library to the test, use cmake with 
+    #    -D "MBED_TEST_LINK_LIBRARIES=mbed-baremetal ext-errorlogging"
+    if (DEFINED MBED_TEST_LINK_LIBRARIES)
+        separate_arguments(MBED_TEST_LINK_LIBRARIES)
+        list(FIND MBED_TEST_LINK_LIBRARIES mbed-baremetal found_position)
+        if(${found_position} LESS 0)
+            list(FIND MBED_TEST_LINK_LIBRARIES mbed-os found_position)
+            if(${found_position} LESS 0)
+                list(APPEND MBED_TEST_LINK_LIBRARIES mbed-os)
```

Comment:
I could be misinterpreting this but it looks like if we don't see `mbed-os` or `mbed-baremetal` in `MBED_TEST_LINK_LIBRARIES` we fall back to linking `mbed-os`? Is this logic really necessary? Should we just pass in what we need to for each call to this macro instead of trying to add default stuff?

Reply:
All green test suites depend on either mbed-os or mbed-baremetal library, and if we are not finding mbed-os or mbed-baremetal in this MBED_TEST_LINK_LIBRARIES  then we make a default configuration to mbed-os as all the test suite can able to run.
And if all greentea tests are going to use the mbed-os library, it is less burden to avoid repeatedly pass the same CMake command-line argument.

@0xc170 Do you have any thoughts

Reply:
lets remove the default then. Let a test to choose what to link. The same way as an app chooses, there is no default, either mbed-os or mbed-baremetal must be linked from an app.

Reply:
This simplifies this snippet to just link the app requested apps with mbed-greentea.